pkg/controllers/rollout/controller.go

```diff
@@ -149,7 +149,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req runtime.Request) (runtim
 
 	// fill out all the default values for CRP just in case the mutation webhook is not enabled.
 	defaulter.SetDefaultsClusterResourcePlacement(&crp)
-
+	// Note: there is a corner case that an override is in-between snapshots (the old one is marked as not the latest while the new one is not created yet)
+	// This will result in one of the override is removed by the rollout controller so the first instance of the updated cluster can experience
+	// a complete removal of the override effect following by applying the new override effect.
+	// This is hard to avoid as the system is eventual consistent and we would recommend users to use updateRun
```

Comment:
For the updateRun, it has the similar issue since cx cannot pick the overrideSnapshot when creating updateRun.
For both cases, cx needs to check the override snapshots before rolling out the new changes and updating the overrides.

Failing this call when it hits this middle state could help the ux.

Reply:
I will add a TODO then